3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```
@@ -18,28 +18,43 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
 function requestData(url) {
-  // TODO return a promise using `fetch()`
+  return new Promise((resolve,reject)=>{
+  fetch(url)
+  .then((response)=> {
+  if (response.ok){
+    return response.json()
+  }else{
+    throw new Error(`Some thing went wrong : ${response.status}`)
```

code style: Since you early return, it would be better to not use an else. You can throw the error if the response is not ok, and then return the response without the else.